lldb-eval/api.h

```diff
@@ -52,11 +55,33 @@ struct ContextVariableList {
   size_t size;
 };
 
+// Context arguments are similar to context variables. The difference is that
+// context arguments maps variable names to types. They are intended to be used
+// for parsing and allow re-using parsed expressions with different values of
+// context variables.
+struct ContextArgument {
+  const char* name;
+  lldb::SBType type;
+};
+
+struct ContextArgumentList {
+  const ContextArgument* data;
+  size_t size;
+};
+
 struct Options {
   bool allow_side_effects = false;
+  ContextArgumentList context_args = {};
   ContextVariableList context_vars = {};
 };
 
+struct ParsedExpression {
+  std::shared_ptr<SourceManager> source;
+  std::unique_ptr<AstNode> tree;
+  lldb::SBTarget target;
```

Comment:
Do we need the target here? It can always be extracted from the value, right?

Reply:
Yes, it can be extracted from the value/frame. Removed.